docs/src/design/stack/crypto_ops.md

```diff
@@ -151,55 +151,60 @@ To keep the degree of the constraints low, a number of intermediate values are u
-    $$\sum_{i=0}^k{\alpha_i \cdot \left(\frac{T_i(x) - T_i(z)}{x - z} + \frac{T_i(x) - T_i(g \cdot z)}{x - g \cdot z} \right)}$$
-where $x$ is the current query to the DEEP composition polynomial for which we are computing the above random linear combination.
-The `RCOMBBASE` instruction computes the numerators $$\alpha_i \cdot (T_i(x) - T_i(z))$$ and $$\alpha_i \cdot (T_i(x) - T_i(g \cdot z))$$ and stores the values in two accumulators $p$ and $r$, respectively. This instruction is specialized to main trace columns i.e. the values $T_i(x)$ are base field elements. The instruction works in combination with the `mem_stream` instruction where it is called 8 times in a row for each call to `mem_stream`.
+## HORNERBASE
+The `HORNERBASE` operation performs $8$ steps of the Horner method for evaluating a polynomial with coefficients over the base field at a point in the quadratic extension field. More precisely, it performs the following update to the accumulator on the stack
+    $$\mathsf{acc}^{'} = (((((((((\mathsf{acc} \cdot \alpha + a_7) \cdot \alpha + a_6) \cdot \alpha + a_5) \cdot \alpha + a_4) \cdot \alpha + a_3) \cdot \alpha + a_2) \cdot \alpha + a_1) \cdot \alpha + a_0$$
+where $a_i$ are the coefficients of the polynomial, $\alpha$ the evaluation point, $\mathsf{acc}$ the current accumulator value and $\mathsf{acc}^{'}$ the updated accumulator value.
 
 The stack for the operation is expected to be arranged as follows:
-- The first $8$ stack elements contain $8$ base field elements $T_0,\cdots , T_7$ representing the values of $T_i(x)$ for the current query $x$ and the current batch of $8$ column values of the main trace for query $x$.
-- The next $2$ elements contain the current value of the accumulator $p$ as a quadratic extension field element.
-- The next $2$ elements contain the current value of the accumulator $r$ as a quadratic extension field element.
-- The next element contains the value of the memory pointer `x_ptr` to the next batch of $8$ column values for query $x$.
-- The next element contains the value of the memory pointer `z_ptr` to the $i$-th OOD evaluations at z and gz i.e. $T_i(z) = (T_i(z)_0, T_i(z)_1)$ and $T_i(gz) = (T_i(gz)_0, T_i(gz)_1)$.
-- The next element contains the value of the memory pointer `a_ptr` to the $i$-th random value $\alpha_i = (\alpha_{i, 0}, \alpha_{i,1})$. The remaining elements of the word are expected to be empty.
+- The first $8$ stack elements contain $8$ base field elements $a_0,\cdots , a_7$ representing the current 8-element batch of coefficients for the polynomial being evaluated.
+- The next $5$ stack elements are irrelevant for the operation and unaffected by it.
+- The next stack element contains the value of the memory pointer `r_ptr` to the evaluation point $\alpha$. The word address containing $\alpha = (\alpha_0, \alpha_1)$ is expected to have layout $[\alpha_0, \alpha_1, 0, 0]$.
```

Comment:
Similar here, I would call this `alpha_ptr` instead of `r_ptr` since we use `alpha` everywhere else.

Comment applies to `HORNEREXT` section too

Reply:
changed